src/pages/api/users/[id]/index.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  editUser,
+  getUser,
+  editPassword,
+} from "../../../../server/db/actions/UserAction";
+import { NextApiRequest, NextApiResponse } from "next";
+import {
+  HTTP_BAD_REQUEST,
+  HTTP_INTERNAL_SERVER_ERROR,
+  HTTP_METHOD_NOT_ALLOWED,
+  HTTP_NOT_FOUND,
+  HTTP_UNAUTHORIZED,
+} from "@/utils/consts";
+import {
+  UserAlreadyExistsException,
+  UserCredentialsIncorrectException,
+  UserDoesNotExistException,
+} from "@/utils/exceptions";
+
+export default async function handler(
+    req: NextApiRequest,
+    res: NextApiResponse,
+) {
+  switch (req.method) {
+    case "GET":
+      const id = String(req.query.id);
+      try {
+        const user = await getUser(id);
+        res.status(200).send({
+          data: user,
+        });
+        return;
+      } catch (e: unknown) {
+        if (e instanceof UserDoesNotExistException) {
+          res.status(HTTP_NOT_FOUND).json({
+            error: (e as Error).message,
+          });
+          return;
+        }
+        res.status(HTTP_INTERNAL_SERVER_ERROR).json({
+          error: (e as Error).message,
+        });
+        return;
+      }
+    
+    case "PUT":
+      await editUserHandler(req, res);
+      break;
+  
+    default:
+      res.status(HTTP_METHOD_NOT_ALLOWED).json({
+        error: `Request method ${req.method} is not allowed`,
+      });
+  }
+  return;
+}
+      
+async function editUserHandler(req: NextApiRequest, res: NextApiResponse) {
+  const { type } = req.query;
+  if (type === "info") {
+    // Editing user profile
+    try {
+      const result = await editUser(req.body);
+      res.status(200).send({
+        data: result,
+      });
+      return;
+    } catch (e: unknown) {
+      if (e instanceof UserAlreadyExistsException) {
+        res.status(HTTP_BAD_REQUEST).json({
+          error: "User with email already exists",
+        });
+        return;
+      }
+      if (e instanceof UserDoesNotExistException) {
+        res.status(HTTP_NOT_FOUND).json({
+          error: (e as Error).message,
+        });
+        return;
+      }
+      res.status(HTTP_INTERNAL_SERVER_ERROR).json({
+        error: (e as Error).message,
+      });
+      return;
```

Comment:
Convert to switch cases for readability:

Here's an example from [SO](https://stackoverflow.com/questions/36332665/how-to-use-instanceof-in-a-switch-statement)
```js
switch (error.constructor) {
    case NotAllowedError:
        return res.send(400);
    case DatabaseEntryNotFoundError:
        return res.send(404);
    default:
        log('Failed to do something async with an unspecified error: ', error);
        return res.send(500);
}
```